frontend/model/contracts/shared/nativeNotification.js

```diff
@@ -9,20 +10,39 @@ export async function requestNotificationPermission (force: boolean = false): Pr
     (force && Notification.permission === 'granted')) {
     return null
   }
-
   try {
-    return await Notification.requestPermission()
+    const permission = await Notification.requestPermission()
+    if (permission === 'granted') {
+      sbp('state/vuex/commit', 'setNotificationGranted', true)
+    } else {
+      sbp('state/vuex/commit', 'setNotificationGranted', false)
+    }
+    return permission
```

Comment:
I do not think these changes are an improvement. They seem to me to be very unnecessary and only add the unnecessary complexity of managing more state for no reason.

It is perfectly fine to call `makeNotification` without checking whether this variable is set, because that function already checks if permissions are granted.

Please remove `setNotificationGranted`, `state.notificationGranted`, and everything associated with it.

Reply:
The reason why I used this state.notificationGranted is because users can not switch Notification permission using the Switch button.
For example, if a user allows permission and switch on, the notifications should be working.
And once he switched off, the notifications should not be working.
But a simple switching off, this can not deny browser notification permission.

That's why it is necessary.

Reply:
@Silver-IT There has to be a better way to implement this.

For example, in `mounted` you could check to see what the value of it is, and then set an ephemeral boolean variable in the NotificationSettings.vue component based on it.

Then it will be sync'd from that point on.

Reply:
> For example, in mounted you could check to see what the value of it is, and then set an ephemeral boolean variable in the NotificationSettings.vue component based on it.

```
  mounted () {
    const permission = Notification?.permission
    if (permission === null) {
      this.disabled = true
    } else {
      this.granted = permission === 'granted' && this.notificationGranted
    }
  },
```

We use `granted` as a init value of the Switch component. Do you mean this?
Actually this value `this.granted` is the final boolean flag represents the if notification will be displayed or not.

Reply:
> We use granted as a init value of the Switch component. Do you mean this?

Yeah that looks good, something like that! 👍

Reply:
`this.notificationGranted` is the Vuex state. So do you agree to use vuex state for this purpose?

Reply:
Oops! Sorry I didn't see that part. No I'm still very much against using vuex for this :P

Why is that part even there? Couldn't it be just:

```js
  mounted () {
    const permission = Notification?.permission
    if (permission === null) {
      this.disabled = true
    } else {
      this.granted = permission === 'granted'
    }
  },
```